src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs

```diff
@@ -0,0 +1,172 @@
+using DotNetNuke.Entities.Portals;
+using Dnn.PersonaBar.Library.Helper;
+using Dnn.PersonaBar.Library.Prompt;
+using Dnn.PersonaBar.Library.Prompt.Models;
+using Dnn.PersonaBar.Pages.Components.Security;
+using Dnn.PersonaBar.Pages.Components.Prompt.Commands;
+
+namespace Dnn.PersonaBar.Pages.Tests
+{
+    [TestFixture]
+    public class GetPageUnitTests
+    {
+        private Mock<ITabController> _tabControllerMock;
+        private Mock<IContentVerifier> _contentVerifierMock;
+        private Mock<ISecurityService> _securityServiceMock;
+
+        [SetUp]
+        public void RunBeforeAnyTest()
+        {
+            _tabControllerMock = new Mock<ITabController>();
+            _securityServiceMock = new Mock<ISecurityService>();
+            _contentVerifierMock = new Mock<IContentVerifier>();
+        }
+
+        [Test]
+        public void Run_GetPageWithValidCommand_ShouldSuccessResponse()
+        {
+            // Arrange
+            var tabId = 21;
+            int testPortalId = 1;
+
+            TabInfo tab = new TabInfo();
+            tab.TabID = tabId;
+            tab.PortalID = testPortalId;
+
+            PortalSettings portalSettings = new PortalSettings();
+            portalSettings.PortalId = testPortalId;
+
+            _securityServiceMock.SetReturnsDefault(true);
+            _tabControllerMock.Setup(t => t.GetTab(tabId, testPortalId)).Returns(tab);
+            _contentVerifierMock.SetReturnsDefault(true);
+
+            IConsoleCommand getCommand = new GetPage(_tabControllerMock.Object, _securityServiceMock.Object, _contentVerifierMock.Object);
+
+            var args = new[] { "get-page", tabId.ToString() };
+            getCommand.Initialize(args, portalSettings, null, tabId);
+
+            // Act
+            var result = getCommand.Run();
+
+            // Assert
+            Assert.IsFalse(result.IsError);
+            Assert.IsNotNull(result.Data);
+            Assert.AreEqual(1, result.Records);
+            Assert.IsFalse(result is ConsoleErrorResultModel);
+
+            _tabControllerMock.Verify(t => t.GetTab(tabId, testPortalId));
+            _securityServiceMock.Verify(s => s.CanManagePage(tabId));
+            _contentVerifierMock.Verify(c => c.IsContentExistsForRequestedPortal(testPortalId, portalSettings, false));
+        }
+
+        [Test]
+        public void Run_GetPageWithValidCommandForNonExistingTab_ShouldErrorResponse()
+        {
+            // Arrange
+            var tabId = 212;
+            int testPortalId = 1;
+
+            TabInfo nullTab = null;
+
+            TabInfo tab = new TabInfo();
+            tab.TabID = tabId;
+            tab.PortalID = testPortalId;
+
+            PortalSettings portalSettings = new PortalSettings();
+            portalSettings.PortalId = testPortalId;
+
+            _tabControllerMock.SetReturnsDefault(nullTab);
+            _securityServiceMock.SetReturnsDefault(true);
+            _contentVerifierMock.SetReturnsDefault(false);
```

Comment:
Common default behaviour can be placed in the `[SetUp]` method to declutter individual tests.

Reply:
Not exactly same for each test